topdown/http_test.go

```diff
@@ -2514,3 +2518,72 @@ var httpSendHelperRules = []string{
 		no_headers = object.remove(resp, ["headers"])
 	}`,
 }
+
+func httpSocket(path string) (net.Listener, error) {
+	sock, err := net.Listen("unix", path)
+	if err != nil {
+		return nil, err
+	}
+
+	// install signal handler to clean up
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
+	go func(c chan os.Signal) {
+		<-c
+		sock.Close()
+		os.Exit(0)
+	}(sigc)
```

Comment:
This is unexpected. 🤔 What happens without it? I'd be cautious about `os.Exit()` anywhere, even more so in a test helper.

Perhaps we can use `t.Cleanup(func() { sock.Close() })` or something like that instead?